File: packages/py/src/airlock/application/ports/run_store.py
"""RunStore port: persists run state so a run can pause and resume (ADR-0004)."""

from typing import Protocol

from airlock.domain.identifiers import RunId
from airlock.domain.run import RunState


class RunStore(Protocol):
    async def save(self, state: RunState) -> None: ...

    async def load(self, run_id: RunId) -> RunState | None: ...

File: packages/ts/src/application/ports/run-store.ts
import type { RunId } from "../../domain/identifiers";
import type { RunState } from "../../domain/run";

/** Persists run state so a run can pause and resume anywhere (see ADR-0004). */
export interface RunStore {
  save(state: RunState): Promise<void>;
  load(runId: RunId): Promise<RunState | null>;
}
